### PR TITLE
haskell.lib.buildStackProject: allow passing in a custom GHC version.

### PR DESCRIPTION
**Pull request by mboes**
Previously, the user could only compile using the default version of GHC
in Nixpkgs. Now this can be changed by setting the `ghc` attribute
appropriately.

### CODE REVIEW

**Comment by YPares** [comment]
Given this recent addition to stack https://github.com/commercialhaskell/stack/commit/4edd98544b05f3e77dcd70feed13693513b7c322
I believe there are two options:
- either use this PR as and modify stack's doc to switch from `buildInputs = [ ghc glpk pcre ];`
to `inherit ghc; buildInputs = [ glpk pcre ];`
- modify this PR so `generic-stack-builder` doesn't handle ghc specifically, as it will just be part of the `buildInputs` (this is how ghc is handled by Stack/Nix right when we give to Stack a list of packages instead of a full `shell.nix`: ghc is just added to the package list before creating the nix expression).

**Comment by obadz** [comment]
cc @peti @bennofs

**Comment by mboes** [comment]
Option (2) breaks backwards compatibility. Which is why I did it the other way. And also for consistency: `buildPythonPackage` et al similarly handle the compiler/runtime choice internally. It's not something the user should be adding to `buildInputs` herself.

**Comment by YPares** [comment]
@m Okay then,
on Stack front, everything is ready. We are waiting for this PR to be merged, given that now it's necessary for us to be able to pass a GHC derivation to the shell.nix file

**Comment by mboes** [comment]
@peti all Stack projects tracking current LTS and using `haskell.lib.buildStackProject` are essentially broken until this PR is merged, due to the default compiler changing to GHC 8 in recent Nixpkgs.

File: doc/languages-frameworks/haskell.md [hunk 1]
```diff
@@ -378,6 +378,23 @@ special options turned on:
...
+Stack choose what GHC version it wants based on the snapshot specified
+in `stack.yaml` (only works with Stack >= 1.1.3):
+
+    {nixpkgs ? import <nixpkgs> { }, ghc ? nixpkgs.ghc}
+
+    with nixpkgs;
+
+    let R = pkgs.R.override { enableStrictBarrier = true; };
+    in
+	haskell.lib.buildStackProject {
```
Comment (peti) [inline doc/languages-frameworks/haskell.md]: This looks like the code snippet is not properly indented. Did you check whether the generated manual looks okay?

Comment (mboes) [inline doc/languages-frameworks/haskell.md]: fixed.

**Pull request merged**